src/pykeen/nn/weighting.py

```diff
@@ -3,34 +3,89 @@
 """Various edge weighting implementations for R-GCN."""
 
 from abc import abstractmethod
+from typing import Union, Optional
 
 import torch
 from class_resolver import Resolver
 from torch import nn
 
+try:
+    from torch_scatter import scatter_add, scatter_max, scatter_softmax
+except ImportError:
+    print("torch-scatter is not installed, attention aggregation won't work. "
+          "Install it here: https://github.com/rusty1s/pytorch_scatter")
+
 __all__ = [
     "EdgeWeighting",
     "InverseInDegreeEdgeWeighting",
     "InverseOutDegreeEdgeWeighting",
     "SymmetricEdgeWeighting",
+    "AttentionEdgeWeighting",
     "edge_weight_resolver",
 ]
 
+def softmax(
```

Comment:
Would https://pytorch.org/docs/stable/generated/torch.sparse.softmax.html be applicable, too?

Reply:
Rather not: 
* it doesn't work on CPU (I have a long error track about not compatible SparseCPU modules of pytorch)
* it is doing smth else than torch_scatter's sparse softmax - ie, there is no way to supply an LongTensor of indices along which we want to compute softmax against a given tensor